lib/sms/sms.c

```diff
@@ -132,62 +162,38 @@ void sms_at_handler(void *context, const char *at_notif)
  */
 static int sms_init(void)
 {
-	char *resp = sms_buf_tmp;
+	uint8_t cnmi_value1;
+	uint8_t cnmi_value2;
+	uint8_t cnmi_value3;
+	uint8_t cnmi_value4;
 	int ret;
 
 	k_work_init(&sms_ack_work, &sms_ack);
 
-	ret = at_params_list_init(&resp_list, AT_SMS_PARAMS_COUNT_MAX);
-	if (ret) {
-		LOG_ERR("AT params error, err: %d", ret);
-		return ret;
-	}
-
 	/* Check if one SMS client has already been registered. */
-	ret = at_cmd_write(AT_SMS_SUBSCRIBER_READ, resp, SMS_BUF_TMP_LEN, NULL);
-	if (ret) {
-		LOG_ERR("Unable to check if an SMS client exists, err: %d",
-			ret);
+	ret = nrf_modem_at_scanf(
```

Comment:
I will try nrf_modem_at_cmd() + sscanf() here to enable mocking in unit tests similarly to at_cmd library.

There is an own CMock stub in modem_jwt test but this doesn't verify variable arguments as those cannot be passed into the stub: https://github.com/nrfconnect/sdk-nrf/blob/main/tests/lib/modem_jwt/src/jwt_test.c#L40